lib/src/misc/simplify.dart

```diff
@@ -0,0 +1,119 @@
+// implementation based on
+// https://github.com/mourner/simplify-js/blob/master/simplify.js
+
+import 'package:latlong2/latlong.dart';
+
+double _getSqDist(
+  LatLng p1,
+  LatLng p2,
+) {
+  final double dx = p1.longitude - p2.longitude;
+  final double dy = p1.latitude - p2.latitude;
+  return dx * dx + dy * dy;
+}
+
+// square distance from a point to a segment
```

Comment:
Probably better to use `///` here

Reply:
It's internal anyway so it doesn't matter too much, but you're right:

```suggestion
/// square distance from a point to a segment
```